torch/utils/data/dataloader.py

```diff
@@ -55,131 +81,84 @@ class DataLoader(object):
-              iterator for the DataLoader is obtained (as in when you call
-              ``enumerate(dataloader,0)``).
-              At this point, the dataset, ``collate_fn`` and ``worker_init_fn`` are passed to each
-              worker, where they are used to access and initialize data based on the indices
-              queued up from the main process. This means that dataset access together with
-              its internal IO, transforms and collation runs in the worker, while any
-              shuffle randomization is done in the main process which guides loading by assigning
-              indices to load. Workers are shut down once the end of the iteration is reached.
-
-              Since workers rely on Python multiprocessing, worker launch behavior is different
-              on Windows compared to Unix. On Unix fork() is used as the default
-              muliprocessing start method, so child workers typically can access the dataset and
-              Python argument functions directly through the cloned address space. On Windows, another
-              interpreter is launched which runs your main script, followed by the internal
-              worker function that receives the dataset, collate_fn and other arguments
-              through Pickle serialization.
-
-              This separate serialization means that you should take two steps to ensure you
-              are compatible with Windows while using workers
-              (this also works equally well on Unix):
-
-              - Wrap most of you main script's code within ``if __name__ == '__main__':`` block,
-                to make sure it doesn't run again (most likely generating error) when each worker
-                process is launched. You can place your dataset and DataLoader instance creation
-                logic here, as it doesn't need to be re-executed in workers.
-              - Make sure that ``collate_fn``, ``worker_init_fn`` or any custom dataset code
-                is declared as a top level def, outside of that ``__main__`` check. This ensures
-                they are available in workers as well
-                (this is needed since functions are pickled as references only, not bytecode).
-
-              By default, each worker will have its PyTorch seed set to
-              ``base_seed + worker_id``, where ``base_seed`` is a long generated
-              by main process using its RNG. However, seeds for other libraies
-              may be duplicated upon initializing workers (w.g., NumPy), causing
-              each worker to return identical random numbers. (See
-              :ref:`dataloader-workers-random-seed` section in FAQ.) You may
-              use :func:`torch.initial_seed()` to access the PyTorch seed for
-              each worker in :attr:`worker_init_fn`, and use it to set other
-              seeds before data loading.
-
     .. warning:: If ``spawn`` start method is used, :attr:`worker_init_fn` cannot be an
```

Comment:
Had another question regarding existing code. This comment seems to suggest we can specify a start method for multiprocessing. Unless I'm missing something it doesn't seem like the DataLoader interface supports a start method.

Reply:
The mp package already provides a way, i.e., `mp.set_start_method`. It sets the start method globally, so DataLoader shouldn't do that automatically. That said, we could let the data loader to take in a mp context, in case that people want to easily use different start methods in the same program.

Reply:
Another option would be to accept a start_method in the dataloader and build an mp context inside the dataloader and set the start method appropriately.

Reply:
Sure sure, we can even let it take in both, but that is really unrelated with this pr.

Reply:
Yes, I completely understand, I just thought I'd mention this for future enhancements.